src/test/firebase-resource-manager.spec.ts

```diff
@@ -0,0 +1,205 @@
+} from "../firebase-resource-manager";
+import * as pollUtils from "../operation-poller";
+import { mockAuth } from "./helpers";
+import { OraWrapper } from "../oraWrapper";
+
+const PROJECT_ID = "the-best-firebase-project";
+const PROJECT_NUMBER = "1234567890";
+const PROJECT_NAME = "The Best Project";
+const PARENT_RESOURCE: ParentResource = { id: "1111111111111", type: ParentResourceType.FOLDER };
+const OPERATION_RESOURCE_NAME_1 = "operations/cp.11111111111111111";
+const OPERATION_RESOURCE_NAME_2 = "operations/cp.22222222222222222";
+
+// TODO(caot): Removed when "Deferred Analytics" and "Deferred Location" are launched
+const TIME_ZONE = "America/Los_Angeles";
+const REGION_CODE = "US";
+const LOCATION_ID = "us-central";
+
+describe("FirebaseResourceManager", () => {
+  describe("createFirebaseProject", () => {
+    let sandbox: sinon.SinonSandbox;
+    let mockOraWrapper: sinon.SinonMock;
+    let apiRequestStub: sinon.SinonStub;
+    let pollOperationStub: sinon.SinonStub;
+    let calledStubs: sinon.SinonStub[];
+
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+      mockAuth(sandbox);
+      mockOraWrapper = sandbox.mock(OraWrapper.prototype);
+      apiRequestStub = sandbox.stub(api, "request");
+      pollOperationStub = sandbox.stub(pollUtils, "pollOperation");
+      calledStubs = [];
+    });
+
+    afterEach(() => {
+      calledStubs.forEach((stub) => expect(stub.calledOnce).to.be.true);
+      mockOraWrapper.verify();
+      sandbox.restore();
+    });
+
+    it("should resolve with project data if it succeeds", async () => {
+      calledStubs.push(
+        _createCloudProjectApiStub().resolves({ body: { name: OPERATION_RESOURCE_NAME_1 } }),
+        _pollCreateCloudProjectOperationStub(OPERATION_RESOURCE_NAME_1).resolves({
+          response: { projectNumber: PROJECT_NUMBER },
+        }),
+        _addFirebaseApiStub().resolves({ body: { name: OPERATION_RESOURCE_NAME_2 } }),
+        _pollAddFirebaseOperationStub(OPERATION_RESOURCE_NAME_2).resolves({
+          response: { projectId: PROJECT_ID, displayName: PROJECT_NAME },
+        })
+      );
+      mockOraWrapper.expects("start").exactly(2);
+      mockOraWrapper.expects("succeed").exactly(2);
+      mockOraWrapper.expects("fail").never();
+
+      expect(await createFirebaseProject(PROJECT_ID, PROJECT_NAME, PARENT_RESOURCE)).to.deep.equal({
+        projectId: PROJECT_ID,
+      });
+    });
+
+    it("should reject if Cloud project creation fails", async () => {
+      const expectedError = new Error("HTTP Error 404: Not Found");
+      calledStubs.push(_createCloudProjectApiStub().rejects(expectedError));
+      mockOraWrapper.expects("start").exactly(1);
+      mockOraWrapper.expects("succeed").never();
+      mockOraWrapper.expects("fail").exactly(1);
+
+      let err;
+      try {
+        await createFirebaseProject(PROJECT_ID, PROJECT_NAME, PARENT_RESOURCE);
+      } catch (e) {
+        err = e;
+      }
+      expect(err.message).to.equals(
```

Comment:
nit: `.to.equal(`. Same goes for the 3 similar occurrences in this file.

Reply:
Done